### PR TITLE
Update API documentation

### PR DESCRIPTION
**Pull request by ayeshLK**
Part of: [#5084](https://github.com/ballerina-platform/ballerina-standard-library/issues/5084)

### CODE REVIEW

File: ballerina/destination.bal [hunk 1]
```diff
@@ -43,14 +54,24 @@ public isolated client class Queue {
...
+    # + return - An `ibmmq:Error` if the operation fails or else `()`
     remote function put(Message message) returns Error? =
     @java:Method {
         'class: "io.ballerina.lib.ibm.ibmmq.Topic"
     } external;
 
+    # Retrieves a message from an IBM MQ topic.
+    # 
+    # + options - Options to control message retrieval
+    # + return - An `ibmmq:Message` if there is any message in the queue, `()` if there
```
Comment (dilanSachi) [inline ballerina/destination.bal:73]:
```suggestion
    # + return - An `ibmmq:Message` if there is any message in the topic, `()` if there 
```

File: ballerina/types.bal [hunk 2]
```diff
@@ -25,16 +34,29 @@ public type QueueManagerConfiguration record {|
     string password?;
 |};
 
+# IBM MQ get message options.
+#
+# + gmOptions - Get message option 
+# + waitInterval - The maximum time (in seconds) that an `get` call waits for a suitable message to
```
Comment (dilanSachi) [inline ballerina/types.bal:40]:
```suggestion
# + waitInterval - The maximum time (in seconds) that n `get` call waits for a suitable message to 
```

Comment (ayeshLK) [reply ballerina/types.bal]: Fixed with [d3ba51763e29ad93627131afb541d17539b784d7](https://github.com/ballerina-platform/module-ballerinax-ibm.ibmmq/pull/12/commits/d3ba51763e29ad93627131afb541d17539b784d7)

File: ballerina/destination.bal (hunk 1 above)
Comment (ayeshLK) [reply ballerina/destination.bal]: Fixed with [d3ba51763e29ad93627131afb541d17539b784d7](https://github.com/ballerina-platform/module-ballerinax-ibm.ibmmq/pull/12/commits/d3ba51763e29ad93627131afb541d17539b784d7)

**Review by Bhashinee — APPROVED** [review]

File: ballerina/Module.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+This module is created with minimal deviation from the IBM MQ java client API to make it easy for the developers who are used to working with the IBM MQ java client.
+ 
+Currently, the following IBM MQ API Classes are supported through this package.
+
+- QueueManager
+- Queue
+- Destination (Queue, Topic)
+- Message
+
+## Samples
```
Comment (dilanSachi) [inline ballerina/Module.md:14]: btw, are we going to add samples later?

File: README.md [hunk 4]
```diff
@@ -1,2 +1,93 @@
-# module-ballerinax-ibm.ibmmq
-Ballerina IBM MQ client connector module.
+# Ballerina `ibm.ibmmq` Library
```
Comment (praneesha) [inline README.md:1]: Should it be `package`? Need to check and fix all occurrences.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,93 @@
...
+[![codecov](https://codecov.io/gh/ballerina-platform/module-ballerinax-ibm.ibmmq/branch/main/graph/badge.svg)](https://codecov.io/gh/ballerina-platform/module-ballerinax-ibm.ibmmq)
+[![Trivy](https://github.com/ballerina-platform/module-ballerinax-ibm.ibmmq/actions/workflows/trivy-scan.yml/badge.svg)](https://github.com/ballerina-platform/module-ballerinax-ibm.ibmmq/actions/workflows/trivy-scan.yml)
+[![GraalVM Check](https://github.com/ballerina-platform/module-ballerinax-ibm.ibmmq/actions/workflows/build-with-bal-test-graalvm.yml/badge.svg)](https://github.com/ballerina-platform/module-ballerinax-ibm.ibmmq/actions/workflows/build-with-bal-test-graalvm.yml)
+[![GitHub Last Commit](https://img.shields.io/github/last-commit/ballerina-platform/module-ballerinax-ibm.ibmmq.svg)](https://github.com/ballerina-platform/module-ballerinax-ibm.ibmmq/commits/main)
+
+The `ballerinax/ibm.ibmmq` library provides an API to connect to an IBM MQ server using Ballerina.
+
+This library is created with minimal deviation from the IBM MQ java client API to make it easy for the developers who are used to working with the IBM MQ java client.
```
Comment (praneesha) [inline README.md:11]: Should be `Java`.

File: README.md [hunk 6]
```diff
@@ -1,2 +1,93 @@
...
+
+Currently, the following IBM MQ API Classes are supported through this package.
```
Comment (praneesha) [inline README.md:13]: Should be `classes`.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,93 @@
...
+
+- QueueManager
```
Comment (praneesha) [inline README.md:15]: Add backticks for classnames.

File: README.md [hunk 8]
```diff
@@ -1,2 +1,93 @@
...
+- Queue
+- Destination (Queue, Topic)
+- Message
+
+## Issues and projects 
+
+Issues and Projects tabs are disabled for this repository as this is part of the Ballerina Standard Library. To report bugs, request new features, start new discussions, view project boards, etc., go to the [Ballerina Standard Library parent repository](https://github.com/ballerina-platform/ballerina-standard-library).
```
Comment (praneesha) [inline README.md:22]: Tab names should be in bold.

File: README.md [hunk 9]
```diff
@@ -1,2 +1,93 @@
...
+- QueueManager
+- Queue
+- Destination (Queue, Topic)
+- Message
+
+## Issues and projects 
+
+Issues and Projects tabs are disabled for this repository as this is part of the Ballerina Standard Library. To report bugs, request new features, start new discussions, view project boards, etc., go to the [Ballerina Standard Library parent repository](https://github.com/ballerina-platform/ballerina-standard-library). 
+
+This repository only contains the source code for the library.
```
Comment (praneesha) [inline README.md:24]: Having this sentence here is confusing.

File: README.md [hunk 10]
```diff
@@ -1,2 +1,93 @@
...
+
+## Build from the source
+
+### Set up the prerequisites
+
+* Download and install Java SE Development Kit (JDK) version 17 (from one of the following locations).
```
Comment (praneesha) [inline README.md:30]: Bullet point should be number 1.

File: README.md [hunk 11]
```diff
@@ -1,2 +1,93 @@
...
+
+2. To run the tests:
+   ```
+   ./gradlew clean test
+   ```
+3. To build the library without the tests:
+   ```
+   ./gradlew clean build -x test
+   ```
+4. To debug library implementation:
```
Comment (praneesha) [inline README.md:57]: Should be `the package implementation`.

File: README.md [hunk 12]
```diff
@@ -1,2 +1,93 @@
...
+   ```
+   ./gradlew clean build -Pdebug=<port>
+   ```
+5. To debug the library with Ballerina language:
```
Comment (praneesha) [inline README.md:61]: Should be `the Ballerina language`.

File: README.md [hunk 13]
```diff
@@ -1,2 +1,93 @@
...
+   ```
+   ./gradlew clean build -PbalJavaDebug=<port>
+   ```
+6. Publish ZIP artifact to the local `.m2` repository:
```
Comment (praneesha) [inline README.md:65]: Add `the`.

File: README.md [hunk 14]
```diff
@@ -1,2 +1,93 @@
...
+   ```
+   ./gradlew clean build publishToMavenLocal
+   ```
+7. Publish the generated artifacts to the local Ballerina central repository:
```
Comment (praneesha) [inline README.md:69]: Central

File: README.md [hunk 15]
```diff
@@ -1,2 +1,93 @@
...
+
+For more information, go to the [contribution guidelines](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md).
+
+## Code of conduct
+
+All the contributors are encouraged to read the [Ballerina Code of Conduct](https://ballerina.io/code-of-conduct).
+
+## Useful links
+
+* For more information go to the [`ibm.ibmmq` library](https://lib.ballerina.io/ballerinax/ibm.ibmmq/latest).
```
Comment (praneesha) [inline README.md:90]: Need a comma.

File: ballerina/Module.md [hunk 16]
```diff
@@ -0,0 +1,14 @@
+## Overview
+
+The `ballerinax/ibm.ibmmq` module provides an API to connect to an IBM MQ server using Ballerina.
```
Comment (praneesha) [inline ballerina/Module.md:3]: Should it be package?

**Pull request merged**